movieland-persistence-jdbc/src/main/java/com/gromoks/movieland/dao/JdbcCountryDao.java

```diff
@@ -0,0 +1,85 @@
+package com.gromoks.movieland.dao;
+
+import com.gromoks.movieland.dao.entity.MovieToCountry;
+import com.gromoks.movieland.dao.mapper.CountryRowMapper;
+import com.gromoks.movieland.dao.mapper.MovieToCountryRowMapper;
+import com.gromoks.movieland.entity.Country;
+import com.gromoks.movieland.entity.Movie;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class JdbcCountryDao implements CountryDao {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    private final MovieToCountryRowMapper movieToCountryRowMapper = new MovieToCountryRowMapper();
+
+    private final CountryRowMapper countryRowMapper = new CountryRowMapper();
+
+    @Autowired
+    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+
+    @Autowired
+    private String getAllMovieToCountrySQL;
+
+    @Override
+    public void enrichSingleMovieByCountries(Movie movie) {
+        log.info("Start to enrich single movie by countries");
+
+        List<Country> countries = getCountryListByMovie(movie);
+
+        if (!Thread.currentThread().isInterrupted()) {
```

Comment:
why check this on dao layer? It should be service

Reply:
I consider it for future implementation